lib/parser.dart

```diff
@@ -313,7 +313,15 @@ class HtmlParser {
       var token = tokenizer.current;
       var newToken = token;
       int type;
+      var attemptCount = 0;
       while (newToken != null) {
+        // Limit the number of attempts to process the current token, and give
+        // up after an arbitrary limit.
+        attemptCount++;
+        if (attemptCount == 100) {
+          parseError(token.span, 'undefined-error');
```

Comment:
We could consider throwing an `AssertionError` instead, or an `UnsupportedError` or `UnimplementedError` because this is more of a known bug that we don't have a fix for, rather than this being a `ParseError`..

I'm assuming that `ParseError` has some special semantics.

Reply:
Good point, I think the `UnimplementedError` is the most appropriate here. PTAL.